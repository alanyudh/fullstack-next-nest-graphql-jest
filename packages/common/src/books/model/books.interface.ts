export interface IBook {
  id: string;
  title: string;
  description: string;
  year: number;
}
