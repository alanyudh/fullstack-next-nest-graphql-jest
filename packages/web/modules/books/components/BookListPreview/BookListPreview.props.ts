import { Book } from '../../models/book.model'

export interface BookListPreviewProps {
  book: Partial<Book>
}
