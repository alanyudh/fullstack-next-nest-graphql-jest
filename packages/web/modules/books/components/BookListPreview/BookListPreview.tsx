import { Card } from 'antd'
import { BookListPreviewProps } from './BookListPreview.props'

const BookListPreview: React.FC<BookListPreviewProps> = ({
  book,
}: BookListPreviewProps) => {
  return (
    <>
      <a data-testid="book-list-anchor" href={`/books/${book.id}`}>
        <Card title={book.title}>{book.shortDescription}</Card>
      </a>
    </>
  )
}

export default BookListPreview
