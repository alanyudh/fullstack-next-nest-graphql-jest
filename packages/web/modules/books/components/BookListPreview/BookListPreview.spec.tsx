import { Book } from '../../models/book.model'
import { render } from '../../../../test/testUtils'
import { BookListPreviewProps } from './BookListPreview.props'
import BookListPreview from './BookListPreview'

describe('BookListPreview', () => {
  let expectedProps: BookListPreviewProps

  beforeEach(() => {
    expectedProps = {
      book: new Book('1', 'title1', 'description1', 2001),
    }
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<BookListPreview {...expectedProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should display title & short', () => {
    const { getByText } = render(<BookListPreview {...expectedProps} />)
    const book = expectedProps.book

    const title = getByText(book?.title)
    const description = getByText(book?.shortDescription)

    expect(title).toBeVisible()
    expect(description).toBeVisible()
  })

  it('should have anchor to book detail', () => {
    const { getByTestId } = render(<BookListPreview {...expectedProps} />)
    const achor = getByTestId('book-list-anchor')

    expect(achor.closest('a').href).toContain(`/books/${expectedProps.book.id}`)
  })
})
