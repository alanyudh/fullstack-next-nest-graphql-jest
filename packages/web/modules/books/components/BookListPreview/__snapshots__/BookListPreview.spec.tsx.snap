// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookListPreview matches snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="book-list-anchor"
    href="/books/1"
  >
    <div
      class="ant-card ant-card-bordered"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            title1
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        description1
      </div>
    </div>
  </a>
</DocumentFragment>
`;
